pkg/sources/job_progress_hook.go

```diff
@@ -25,6 +26,10 @@ func WithUnitHookCache(cache *lru.Cache[string, *UnitMetrics]) UnitHookOpt {
 	return func(hook *UnitHook) { hook.metrics = cache }
 }
 
+func OnUnitHookFinish(f func(UnitMetrics)) UnitHookOpt {
+	return func(hook *UnitHook) { hook.finishFunc = f }
+}
```

Comment:
I'm not fully convinced this is a good solution and am open to alternatives.

Another idea is to spawn a goroutine to periodically flush the hook and look for metrics that are finished to write out.

Reply:
What don't you like about it?

Reply:
It's a specific option for this specific hook. If there are other hooks that similarly need to be flushed, the solution doesn't scale. I think that's acceptable for now since there's just this one hook, but a system that knows about a `Flush` method and calls it might be nice.